From: Sagi Grimberg <sagi@grimberg.me>
Date: Tue, 15 Aug 2017 22:20:38 +0300
Subject: iwcm: Don't allocate iwcm workqueue with WQ_MEM_RECLAIM
Patch-mainline: v4.14-rc1
Git-commit: 75215e5bb22c9c502e47398bdc35c247b1469809
References: bsc#1046306 FATE#322942

Its very likely that iwcm work execution will yield memory
allocations (for example cm connection request).

Reported-by: Steve Wise <swise@opengridcomputing.com>
Reviewed-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Sagi Grimberg <sagi@grimberg.me>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/core/iwcm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/core/iwcm.c
+++ b/drivers/infiniband/core/iwcm.c
@@ -1177,7 +1177,7 @@ static int __init iw_cm_init(void)
 		pr_err("iw_cm: couldn't init iwpm\n");
 	else
 		rdma_nl_register(RDMA_NL_IWCM, iwcm_nl_cb_table);
-	iwcm_wq = alloc_ordered_workqueue("iw_cm_wq", WQ_MEM_RECLAIM);
+	iwcm_wq = alloc_ordered_workqueue("iw_cm_wq", 0);
 	if (!iwcm_wq)
 		return -ENOMEM;
 
